solr/core/src/test/org/apache/solr/response/TestRawResponseWriter.java

```diff
@@ -162,6 +176,33 @@ public void testStructuredDataViaBaseWriters() throws IOException {
     assertEquals("test", out.getVal(0));
     assertEquals("foo", out.getName(1));
     assertEquals("bar", out.getVal(1));
+
+    // cbor
+
+    byte[] cborBytes = serializeToCbor(json.getBytes());
+    assertEquals(25, cborBytes.length);
+    LongAdder numberOfObjectsInResponse = new LongAdder();
+    new CborLoader(null, (document) -> numberOfObjectsInResponse.increment())
+        .stream(new ByteArrayInputStream(cborBytes));
+    assertEquals(1, numberOfObjectsInResponse.intValue());
+  }
+
+  private byte[] serializeToCbor(byte[] is) throws IOException {
+    ByteArrayOutputStream baos;
```

Comment:
It's better to define vars just-in-time (later) and closer to the logical flow of usage, not all up front like this is C code.